lang/php/Portfile

```diff
@@ -1356,6 +1356,22 @@ subport ${php}-openssl {
     
     configure.args-append   --with-kerberos=${prefix} \
                             --with-openssl=${prefix}
+
+    if {[vercmp ${branch} 5.6] <= 0} {
+        depends_lib-replace     path:lib/libssl.dylib:openssl port:openssl10
+        depends_build-append    port:pkgconfig
+
+        # This patch is for reordering -I and -L flags during the build to make
+        # openssl-1.0 directories appear before others
+        patchfiles-append       patch-${php}-ext-openssl-config.m4.diff
+
+        post-patch {
+            reinplace "s|@PREFIX@|${prefix}|g" ${worksrcpath}/ext/openssl/config.m4
+        }
+
+        configure.args-replace  --with-openssl=${prefix} --with-openssl=shared
+        configure.env           PKG_CONFIG_PATH=${prefix}/lib/openssl-1.0/pkgconfig
```

Comment:
Hmmm. I am increasingly wondering if it makes sense to put logic like this in multiple places, in ports that need tweaks to build against openssl 1.0. It just seems to me we will end up repeating similar logic in multiple places, at which point I think we should think about putting it into something like an ssl PortGroup which can then handle setting up ports to correctly build agains openssl 1.0, or 1.1 (or, going forward 1.x).

Reply:
A port group will make sense. But as there are only few ports need openssl10 (17 ports for now), I'm fine with copy-and-paste :)

Reply:
17 is for me already way more than I would think justifies a PG. repeating code even once isn’t a good idea ;)

Reply:
I'm persuaded :) Will have a look.

Reply:
Have you had chance to look into an ssl portgroup ?

Reply:
I've got a proof-of-concept. Will push it after some testing.